### ISSUE TITLE
Bug: Windows case-sensitivty causes eslint to incorrectly flag included TypeScript files as not included

### ISSUE THREAD

**Issue by aaronmcdavid** [eslint/eslint#16773]
### Environment

Node version: v18.12.1
npm version: v9.2.0
Local ESLint version: v8.31.0 (Currently used)
Global ESLint version: Not found
Operating System: win32 10.0.22621

git version 2.39.0.windows.1

### What parser are you using?

@typescript-eslint/parser

### What did you do?

On Windows, with the below invocation and configuration, eslint generates incorrect error messages if the folder in which it is run has case-sensitivity turned on.  It runs properly if case-sensitivity is turned off (the Windows default), either when cloning or if it is turned off later.

Invocation:
```
npx eslint ./src --ext .tsx
```
.tsconfig.eslint.json
```
{
  "include": ["**/*"]
}
```
Relevant sections of .eslintrc.json
```
    "files": ["*.ts", "*.tsx"],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
      "ecmaVersion": "latest",
      "project": ["tsconfig.eslint.json"]
    }
```
Command to enable case-sensitivity (clone into the folder after this is run)
```
fsutil.exe file setCaseSensitiveInfo "«Path»" enable
```
Command to disable case-sensitivity on folder and all subfolders (PowerShell)
```
fsutil.exe file setCaseSensitiveInfo "«Path»" disable
Get-ChildItem -Path "«Path»" -Recurse -Directory | ForEach-Object { fsutil.exe file setCaseSensitiveInfo "$($_.FullName)" disable }
```

### What did you expect to happen?

When project is cloned into a folder with case-sensitivity turned off, the error messages indicated below do not occur.  The error messages also do not occur if the project is cloned into a folder with case-sensitivity turned on, and case-sensitivity is subsequently turned off in the method described above.

### What actually happened?

When case-sensitivity turned on, eslint emits the following error for every .tsx file (specific example given):

```
C:\test\frontend\src\_components\Global\Input\StyledDatePickerRenderInput.tsx
  0:0  error  Parsing error: ESLint was configured to run on `<tsconfigRootDir>/src\_components\Global\Input\StyledDatePickerRenderInput.tsx` using `parserOptions.project`: <tsconfigRootDir>/tsconfig.json
However, that TSConfig does not include this file. Either:
- Change ESLint's list of included files to not include this file
- Change that TSConfig to include this file
- Create a new TSConfig that includes this file and include it in your parserOptions.project
See the TypeScript ESLint docs for more info: https://typescript-eslint.io/linting/troubleshooting##i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by nzakas** [comment]
This is actually an issue with [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint) rather than ESLint itself. Please open an issue there.
